modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/download_link_query.rb

```diff
@@ -52,30 +52,31 @@ def call(user:, file_link:)
                 headers: { 'Authorization' => "Bearer #{token.access_token}" }
               )
 
-              if (300..399).include?(response.status)
-                ServiceResult.success(result: response.headers['Location'])
-              else
-                handle_errors(response)
-              end
+              handle_errors(response)
             end
           end
 
           private
 
           def handle_errors(response)
+            json = MultiJson.load(response.body.to_s, symbolize_keys: true)
```

Comment:
If `response` is a `HTTPX::ErrorResponse` then we will have an issue here because there is no `HTTPX::ErrorResponse#body`.
So, I suggest to postpone json parsing to the moment we are sure body is present. And then json can be received from response like: `response.json(symbolize_names: true)`.

Reply:
very good catch

Reply:
Can we somehow write a test for it? you know, have a webmock stub, that is converted by httpx into an `ErrorResponse`? I will have a look.

Reply:
Have a look at this one: https://github.com/opf/openproject/blob/dev/modules/storages/spec/contracts/storages/storages/shared_contract_examples.rb#L180

But actually there is a problem with it, because it raises an error while in reality it does not. I will think about it.

Reply:
But actually I think we should do it with WebMock: https://github.com/bblimke/webmock?tab=readme-ov-file#raising-timeout-errors
Then whatever library we use will behave accordingly(raise a timeout error or return an appropriate error object)

Reply:
I solved it now with mocking the `HTTPX::get` method. This has the advantage, that I can have those tests right next to all the other unit tests of the query.

Inside a VCR context we cannot use webmock and `.to_timeout`